### PR TITLE
Various improvements to our documentation

### LINKED ISSUE #196: Document default config values
Currently the default values are not included in our MkDocs-based documentation. I'm not sure **how** to include them, so I've asked this question upstream:

https://github.com/pawamoy/mkdocstrings/issues/177

### LINKED ISSUE #447: DOC: add the default value of config options in the documentation
It would be nice to add the default valuesof configuration options in the documentation. 
This way when we look at the definition of a particular configuration option in the doc we know if we have to override its default value or not (instead of looking for it in config.py).

### PR DESCRIPTION
**Pull request by hoechenberger**
- Use latest mkdocstrings-python so we get default parameter values in the docs
- Reduce table row height in example docs
- Fix icons used in the report download buttons in the example gallery
- Update the URL displayed by the browser when clicking on an anchor link in the docs, so it will be easier to share anchors
- Change section order on example pages
- Change order of documentation pages, so that the next page after the landing page is installation & getting started
- Fix missing / broken links to rendered examples in our gallery
- Some more minor stuff
    
Closes #196
Closes #447

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -83,16 +84,20 @@ plugins:
...
-                setup_commands:
-                    - import os
-                    - os.environ["MNE_BIDS_STUDY_CONFIG"] = "./source/doc-config.py"
-                    - import sys
-                    - sys.path.insert(0, "..")
+                # setup_commands:
+                #     - import os
+                #     - os.environ["MNE_BIDS_STUDY_CONFIG"] = "./source/doc-config.py"
+                #     - import sys
+                #     - sys.path.insert(0, "..")
```
Comment (hoechenberger) [inline docs/mkdocs.yml:91]:
```suggestion
```

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -83,16 +84,20 @@ plugins:
...
                 rendering:
                     show_source: true
                     show_if_no_docstring: false
                     show_root_heading: true
-                    show_root_full_path: false
+                    # show_root_full_path: false
```
Comment (hoechenberger) [inline docs/mkdocs.yml:96]:
```suggestion
                    show_root_full_path: false
```

File: tests/requirements.txt [hunk 3]
```diff
@@ -4,8 +4,9 @@ mkdocs
 mkdocs-material
 mkdocs-material-extensions
 mkdocs-macros-plugin
-mkdocstrings >= 0.16
-pytkdocs >= 0.11
+mkdocstrings-python
+jinja2 < 3.1.0  # https://twitter.com/readthedocs/status/1507388916013314048
+black  # docstring formatting
```
Comment (hoechenberger) [inline tests/requirements.txt:9]:
```suggestion
black  # function signature formatting
```

File: docs/source/governance.md [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
 ## Steering Council
 
 - Alex Gramfort
-- Richard Hoechenberger
+- Richard Höchenberger
 
 ## Institutional Partners
 
-- [Institut national de recherche en informatique et en automatique](https://www.inria.fr/)
+- [Institut national de recherche en informatique et en automatique (INRIA)](https://www.inria.fr/)
```
Comment (agramfort) [inline docs/source/governance.md:13]:
```suggestion
- [Inria](https://www.inria.fr/)
```

**Comment by hoechenberger** [comment]
Built docs: https://output.circle-artifacts.com/output/job/64068c87-a04d-436b-97ae-215268ad66ba/artifacts/0/site/index.html

@agramfort merge if happy!

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
Beautiful 🤩

**Pull request merged**